connectors/aws_rds_oracle/readme.md

```diff
@@ -0,0 +1,97 @@
+
+## Features
+
+-   Direct connection to AWS RDS Oracle database
+-   Incremental data extraction based on the `LAST_UPDATED` column
+-   Support for primary key identification
+-   Easy configuration via JSON file
+
+## Configuration file
+
+The connector requires the following configuration parameters in `configuration.json`:
+
+```json
+{  "host": "<YOUR_RDS_ORACLE_HOST>",  "port": "1521",  "service_name": "<YOUR_SERVICE_NAME>",  "user": "<YOUR_DB_USERNAME>",  "password": "<YOUR_DB_PASSWORD>"}
+```
+
+Note: Ensure that the `configuration.json` file is not checked into version control to protect sensitive information.
+
+## Requirements file
+
+The connector requires the `oracledb` package to connect to Oracle databases.
+
+```
+oracledb==3.3.0
+```
+
+Note: The `fivetran_connector_sdk` and `requests` packages are pre-installed in the Fivetran environment.To avoid dependency conflicts, do not declare them in your `requirements.txt`.
+
+## Authentication
+
+This connector uses username and password authentication to connect to Oracle.The credentials are specified in the configuration file and passed to the `oracledb.connect()` function (see the `connect_oracle()` function).
+
+To set up authentication:
+
+-   Create an Oracle user with appropriate permissions to access the required tables.
+-   Provide the username and password in the `configuration.json` file.
+-   Ensure the user has `SELECT` permissions on the tables that need to be synced.
+
+## Pagination
+
+The connector handles pagination by filtering rows with `_build_incremental_query()` and processing results in batches through `_iterate_records()`.The Oracle cursor uses `cursor.fetchmany()` with a configured `arraysize`, so the connector efficiently streams rows without requiring manual page tokens.
+
+## Data handling
+
+The connector handles data as follows (see the `update()` function):
+
+1.  Connects to the Oracle database using the provided configuration.
+2.  Retrieves the last sync timestamp from state (or uses a default date for the initial sync).
+3.  Executes a SQL query to fetch records modified after the last sync timestamp.
+4.  Transforms each database record into the target schema format.
+5.  Performs upsert operations for each record.
+6.  Maintains state with the latest processed timestamp.
+
+## Error handling
+
+`validate_configuration()` verifies all required configuration keys before any work begins and raises descriptive errors when values are missing.`connect_oracle()` catches invalid port values, and the `update()` function logs connection failures with `log.severe()` before re-raising the exception.During syncs, the connector wraps Oracle interactions in a `try/finally` block to ensure the database connection is closed, and it checkpoints progress with `op.checkpoint()` to prevent re-processing if an error occurs mid-batch.
```

Comment:
```suggestion
`validate_configuration()` verifies all required configuration keys before any work begins and raises descriptive errors when values are missing.`connect_oracle()` catches invalid port values, and the `update()` function logs connection failures with `log.severe()` before re-raising the exception. During syncs, the connector wraps Oracle interactions in a `try/finally` block to ensure the database connection is closed, and it checkpoints progress with `op.checkpoint()` to prevent re-processing if an error occurs mid-batch.
```